src/openassetio-core/include/openassetio/hostApi/Manager.hpp

```diff
@@ -665,7 +665,94 @@ class OPENASSETIO_CORE_EXPORT Manager {
+   * "TraitsData". The conceptual meaning of the call is:
+   *
+   * "I have this reference you gave me, and I would like to register
+   * a new entity to it with the traits I told you about before. I
+   * trust that this is ok, and you will give me back the reference
+   * that represents the result of this."
+   *
+   * It is up to the manager to understand the correct result for the
+   * particular trait set in relation to this reference. For example,
+   * if you received this reference in response to browsing for a
+   * target to `kWriteMultiple` and the traits of a
+   * `ShotSpecification`s, then the Manager should have returned you
+   * a reference that you can then register multiple
+   * `ShotSpecification` entities to without error. Each resulting
+   * entity reference should then reference the newly created Shot.
+   *
+   * @warning All supplied TraitsDatas should have the same trait
+   * sets. If you wish to register different "types" of entity, they
+   * need to be registered in separate calls.
+   *
+   * @warning When registering files, it should never be assumed
+   * that the resulting @ref entity_reference will resolve to the
+   * same path. Managers may freely relocate, copy, move or rename
+   * files as part of registration.
+   *
+   * @param entityReferences Entity references to register_ to.
+   *
+   * @param entityTraitsDatas The data to register for each entity.
+   * NOTE: All supplied instances should have the same trait set.
+   *
+   * @param context Context The calling context.
+   *
+   * @param successCallback Callback that will be called for each
+   * successful registration of an entity reference. It will be given
+   * the corresponding index of the entity reference in
+   * `targetEntityRefs` along with an updated reference to use for
+   * future interactions with the resulting new entity. The callback
+   * will be called on the same thread that initiated the call to
+   * `register`.
+   *
+   * @param errorCallback Callback that will be called for each
+   * failed registration of an entity reference. It will be given the
+   * corresponding index of the entity reference in `entityRefs`
+   * along with a populated @fqref{BatchElementError}
+   * "BatchElementError" (see @fqref{BatchElementError.ErrorCode}
+   * "ErrorCodes"). The callback will be called on the same thread
+   * that initiated the call to `register`.
+   *
+   * @return None
+   *
+   * @exception `std::out_of_range` If `entityReferences` and
+   * `entityTraitsDatas` are not lists of the same length.
+   *
+   * @exception `std::invalid_argument` If all `entityTraitsDatas` do
+   * not share the same trait set.
+   *
+   * Other exceptions may be raised for fatal runtime errors, for
+   * example server communication failure.
+   *
+   * @see @fqref{TraitsData} "TraitsData"
+   * @see @ref preflight
+   */
+  // NOLINTNEXTLINE(readability-identifier-naming)
+  void register_(const EntityReferences& entityReferences,
```

Comment:
[question] I just realized, there is a slightly unfortunate collision here, this is our google clang-format convention for class data members. Can we get away with that for now until we think of a new name?

Reply:
Yeah this should really just be a temporary name.  The new name may be influenced by the conveniences we choose to add, e.g. `registerOne` vs. `registerMultiple` etc.